service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/IssuingDetailsApprovalDateBlockPrePopulator.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.fpl.service.orders.prepopulator.question;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.time.LocalDate;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock.APPROVAL_DATE;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class IssuingDetailsApprovalDateBlockPrePopulator implements QuestionBlockOrderPrePopulator {
```

Comment:
this prepopulator when Approval date is there, is that the intention?

Reply:
Yes, I have an AC that asked for this to be added 
```On the approval/issuing details page, pre-populate the approval date with current date and allow user to edit (this should behave the same for all order types)```

Reply:
ah sorry, got confused by the name. Probably ApprovalDateBlockPrePopulator?